gnovm/stdlibs/std/crypto.gno

```diff
@@ -41,6 +43,29 @@ func DecodeBech32(addr Address) (string, [20]byte, bool) {
 	return prefix, convertTo20Byte(bz), true
 }
 
+func PubkeyToAddress(publicKey string) (Address, error) {
+	_, bz, err := decodeAndConvert(publicKey)
+	if err != nil {
+		return "", err
+	}
+	//XXX: the bytes we recover here are in the form:
+	/// /tm.KEY_TYPE#\n!ACTUAL_KEY_BYES
+	// so we need to split the string in order to get the actual key bytes
+	// we also ensure that the type of the key is supported right now just PubKeySecp256k1
+	parts := strings.Split(string(bz), "#\n!")
+	keyType := parts[0]
+	keyBytes := []byte(parts[1])
+
+	if !strings.Contains(keyType, "/tm.PubKeySecp256k1") {
```

Comment:
Based on the comment above here you can directly check if `keyType` has a prefix of `"/tm.PubKeySecp256k1"` by slicing.